plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -655,13 +646,24 @@ public ConnectorTableMetadata getTableMetadata(ConnectorSession session, Connect
 
     private List<ColumnMetadata> getTableColumnMetadata(MetadataEntry metadataEntry, ProtocolEntry protocolEntry)
     {
-        Map<String, String> columnComments = getColumnComments(metadataEntry);
-        Map<String, Boolean> columnsNullability = getColumnsNullability(metadataEntry);
-        Map<String, String> columnGenerations = getGeneratedColumnExpressions(metadataEntry);
-        List<ColumnMetadata> columns = getColumns(metadataEntry, protocolEntry).stream()
-                .map(column -> getColumnMetadata(column, columnComments.get(column.getBaseColumnName()), columnsNullability.getOrDefault(column.getBaseColumnName(), true), columnGenerations.get(column.getBaseColumnName())))
-                .collect(toImmutableList());
-        return columns;
+        DeltaLakeTable table = DeltaLakeTable.builder(metadataEntry, protocolEntry).build();
+        ImmutableList.Builder<ColumnMetadata> columns = ImmutableList.builder();
+        for (DeltaLakeColumnHandle columnHandle : getColumns(metadataEntry, protocolEntry)) {
+            String name = columnHandle.getBaseColumnName();
+            ColumnMetadata columnMetadata = switch (columnHandle.getColumnType()) {
+                case REGULAR, PARTITION_KEY -> {
+                    DeltaLakeColumn column = table.findColumn(name);
+                    yield getColumnMetadata(columnHandle, column.comment(), column.nullable(), column.generationExpression());
+                }
+                case SYNTHESIZED -> ColumnMetadata.builder()
```

Comment:
I'm not sure why we have this "special" way of building `ColumnMetadata` instances when `getColumnMetadata(...)` can already cope with synthetized columns.

Either use `getColumnMetadata` as before or add a comment.